docs/rules/no-for-loop.md

```diff
@@ -0,0 +1,27 @@
+# Do not use for loop that can be replaced with a for-of loop
+
+There's no reason to use old school for loops anymore for the common case. You can instead use for-of loop (with `.entries()` if you need to access the index).
+
```

Comment:
I think the docs here should mention something about this preventing off-by-one errors.

From the issue:

> Off-by-one errors is one of the most common bugs in software. [Swift actually removed this completely from the language.](https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md)